### PR TITLE
Simplify Stream Operations

### PR DESCRIPTION
**Pull request by jjbuchan**
Creating a PR for george's suggestion

### CODE REVIEW

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java [hunk 1]
```diff
@@ -119,27 +120,15 @@ public void delete(UUID operatorId) {
...
-
+        // create a hashmap of (agenttype/monitortype:translations)
+        .collect(Collectors.toMap(op -> new MonitorTranslationOperator().setAgentType(op.getAgentType()).setMonitorType(op.getMonitorType()),
+            op -> List.of(op.getTranslatorSpec().info()),
+            ListUtils::union))
+        // convert the map into a stream of details
+        .entrySet().stream().map(entry ->
+            new MonitorTranslationDetails().setAgentType(entry.getKey().getAgentType())
+            .setMonitorType(entry.getKey().getMonitorType())
+            .setTranslations(entry.getValue()))
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java:131]:
```suggestion
        .entrySet().stream().map(entry ->
            entry.getKey().setTranslations(entry.getValue()))
```

Comment (jjbuchan) [reply src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java:131]: trying this locally and it's not right.  will tweak again once i see what's up

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java [hunk 2]
```diff
@@ -119,27 +120,15 @@ public void delete(UUID operatorId) {
...
-                                .setMonitorType(monitorEntry.getKey())
-                                .setTranslations(monitorEntry.getValue())))),
-
-                // then assign the agent type to each MonitorTranslationDetails
-                agentMap -> agentMap.entrySet().stream()
-                    .flatMap(agentEntry -> agentEntry.getValue()
-                        .map(detail -> detail.setAgentType(agentEntry.getKey())))))
-
+        // create a hashmap of (agenttype/monitortype:translations)
+        .collect(Collectors.toMap(op -> new MonitorTranslationOperator().setAgentType(op.getAgentType()).setMonitorType(op.getMonitorType()),
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java:124]:
```suggestion
        .collect(Collectors.toMap(op -> new MonitorTranslationDetails().setAgentType(op.getAgentType()).setMonitorType(op.getMonitorType()),
```

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java [hunk 3]
```diff
@@ -119,27 +120,15 @@ public void delete(UUID operatorId) {
...
+            op -> List.of(op.getTranslatorSpec().info()),
+            ListUtils::union))
```
Comment (jjbuchan) [inline src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java:126]: From what I can tell it looks like the expected order does remain intact after this step.

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java (hunk 2 above)
Comment (GeorgeJahad) [reply src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java:124]: +1

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java (hunk 1 above)
Comment (GeorgeJahad) [reply src/main/java/com/rackspace/salus/monitor_management/services/MonitorContentTranslationService.java:131]: +1

**Pull request merged**